invoices/invoice_expiry_watcher.go

```diff
@@ -128,48 +129,23 @@ func (ew *InvoiceExpiryWatcher) prepareInvoice(
 }
 
 // AddInvoices adds multiple invoices to the InvoiceExpiryWatcher.
-func (ew *InvoiceExpiryWatcher) AddInvoices(
-	invoices map[lntypes.Hash]*channeldb.Invoice) {
-
-	invoicesWithExpiry := make([]*invoiceExpiry, 0, len(invoices))
-	for paymentHash, invoice := range invoices {
-		newInvoiceExpiry := ew.prepareInvoice(paymentHash, invoice)
-		if newInvoiceExpiry != nil {
-			invoicesWithExpiry = append(
-				invoicesWithExpiry, newInvoiceExpiry,
-			)
-		}
-	}
-
-	if len(invoicesWithExpiry) > 0 {
+func (ew *InvoiceExpiryWatcher) AddInvoices(invoices []*invoiceExpiry) {
```

Comment:
nit: should the godoc explain that all entries must be non-nil?

Reply:
Decided to instead make sure that even if we push `nil` entries, they won't be used (see select at the end)